lib/oban/validation.ex

```diff
@@ -125,6 +125,12 @@ defmodule Oban.Validation do
 
   defp validate_type({:custom, fun}, key, val) when is_function(fun, 1) do
     with {:error, message} <- fun.(val) do
+      message =
+        case message do
+          %ArgumentError{message: message} -> message
+          other -> other
+        end
+
```

Comment:
This could work for any exception struct:

```suggestion
    with {:error, error} <- fun.(val) do
      message = if is_exception(error), do: error.message, else: error

```